### PR TITLE
Kafka settings updates

### PR DESCRIPTION
**Pull request by emord**
@dimagi/scale-team 

https://kafka-python.readthedocs.io/en/master/apidoc/KafkaConsumer.html

### CODE REVIEW

File: corehq/apps/change_feed/consumer/feed.py [hunk 1]
```diff
@@ -132,6 +131,8 @@ def _get_consumer(self, timeout, auto_offset_reset='smallest'):
             'bootstrap_servers': settings.KAFKA_BROKERS,
             'consumer_timeout_ms': timeout,
             'auto_offset_reset': auto_offset_reset,
+            'api_version': settings.KAFKA_API_VERSION,
+            'enable_auto_commit': False,
```
Comment (emord) [inline corehq/apps/change_feed/consumer/feed.py]: we don't need auto commit because we don't use kafka to store offsets

File: settings.py [hunk 2]
```diff
@@ -857,6 +857,7 @@
 ICDS_SMS_INDICATOR_DOMAINS = []
 
 KAFKA_BROKERS = ['localhost:9092']
+KAFKA_API_VERSION = (0, 8, 2)
```
Comment (emord) [inline settings.py]: This stops it from having to infer

File: corehq/apps/change_feed/consumer/feed.py [hunk 3]
```diff
@@ -65,8 +65,7 @@ def iter_changes(self, since, forever):
         if since is not None and (not isinstance(since, dict) or not since):
             raise ValueError("'since' must be None or a topic offset dictionary")
 
-        # in milliseconds, -1 means wait forever for changes
-        timeout = -1 if forever else MIN_TIMEOUT
+        timeout = float('inf') if forever else MIN_TIMEOUT
```
Comment (emord) [inline corehq/apps/change_feed/consumer/feed.py]: float('inf') is default. not clear if -1 still works, but i think it does

File: corehq/apps/change_feed/producer.py [hunk 4]
```diff
@@ -64,7 +64,10 @@ def kafka(self):
     def producer(self):
         if self._producer is None and not self._has_error:
             if self.kafka is not None:
-                self._producer = SimpleProducer(self._kafka)
+                self._producer = SimpleProducer(
+                    self._kafka, async_send=False, req_acks=SimpleProducer.ACK_AFTER_LOCAL_WRITE,
+                    sync_fail_on_error=True
```
Comment (emord) [inline corehq/apps/change_feed/producer.py]: pulled from synchronous example https://kafka-python.readthedocs.io/en/master/simple.html#synchronous-mode

**Review by snopoke — APPROVED** [review]

**Pull request merged**